pkg/app/mongodb.go

```diff
@@ -0,0 +1,220 @@
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/helm"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/log"
+)
+
+type MongoDB struct {
+	cli       kubernetes.Interface
+	namespace string
+	dbname    string
+	username  string
+	password  string
+	name      string
+	chart     helm.ChartInfo
+}
+
+func NewMongoDB(name string) App {
+	return &MongoDB{
+		dbname:   "admin",
+		username: "root",
+		name:     name,
+		chart: helm.ChartInfo{
+			Release:  name,
+			RepoURL:  helm.StableRepoURL,
+			Chart:    "mongodb",
+			RepoName: helm.StableRepoName,
+			Version:  "7.4.6",
+			Values: map[string]string{
+				"replicaSet.enabled":  "true",
+				"image.repository":    "kanisterio/mongodb",
+				"image.tag":           "0.22.0",
+				"mongodbRootPassword": "secretpassword",
+			},
+		},
+	}
+}
+
+func (mongo *MongoDB) Init(ctx context.Context) error {
+	// Instantiate Client SDKs
+	cfg, err := kube.LoadConfig()
+	if err != nil {
+		return err
+	}
+	mongo.cli, err = kubernetes.NewForConfig(cfg)
+
+	return err
+}
+
+func (mongo *MongoDB) Install(ctx context.Context, namespace string) error {
+	mongo.namespace = namespace
+
+	cli := helm.NewCliClient(helm.V3)
+	log.Print("Appdig repo for the application.", field.M{"app": mongo.name})
+
+	err := cli.AddRepo(ctx, mongo.chart.RepoName, mongo.chart.RepoURL)
+	if err != nil {
+		return err
+	}
+
+	log.Print("Installing application using helm.", field.M{"app": mongo.name})
+	err = cli.Install(ctx, fmt.Sprintf("%s/%s", mongo.chart.RepoName, mongo.chart.Chart), mongo.chart.Version, mongo.name, mongo.namespace, mongo.chart.Values)
+
+	if err != nil {
+		return err
+	}
+	log.Print("Application was installed successfully.", field.M{"app": mongo.name})
+	return nil
+}
+
+func (mongo *MongoDB) IsReady(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
+	defer cancel()
+
+	err := kube.WaitOnStatefulSetReady(ctx, mongo.cli, mongo.namespace, fmt.Sprintf("%s-mongodb-primary", mongo.name))
+	if err != nil {
+		return false, err
+	}
+
+	err = kube.WaitOnStatefulSetReady(ctx, mongo.cli, mongo.namespace, fmt.Sprintf("%s-mongodb-secondary", mongo.name))
+	if err != nil {
+		return false, err
+	}
+
+	err = kube.WaitOnStatefulSetReady(ctx, mongo.cli, mongo.namespace, fmt.Sprintf("%s-mongodb-arbiter", mongo.name))
+	if err != nil {
+		return false, err
+	}
+
+	log.Print("Application is ready.", field.M{"app": mongo.name})
+	return true, nil
+}
+
+func (mongo *MongoDB) Object() crv1alpha1.ObjectReference {
+	return crv1alpha1.ObjectReference{
+		Kind:      "StatefulSet",
+		Name:      fmt.Sprintf("%s-mongodb-primary", mongo.name),
+		Namespace: mongo.namespace,
+	}
+}
+
+func (mongo *MongoDB) Uninstall(ctx context.Context) error {
+	cli := helm.NewCliClient(helm.V3)
+
+	log.Print("Uninstalling application.", field.M{"app": mongo.name})
+	err := cli.Uninstall(ctx, mongo.name, mongo.namespace)
+	if err != nil {
```

Comment:
We can just return `errors.Wrapf(err)`here. If error is nil errors.Wrapf() also return nil

Reply:
yeah, I understand, but if there is an error I wanted to print when did we encounter the error. Do you think `errors.Wrapf(err)` is better idea.

Reply:
We are anyways returning `errors.Wrapf(err, "Error while uninstalling the application.")` on line no 134. What I meant is we don't need to check `if err != nil` if we are using `errors.Wrapf()` function